Runnable Embedded GlassFish improvements (stop on error, print message at the end)

OndroMih opened the pull request:
* stop automatically on error when executing commands
* print Welcome message at the beginning
* print message after shutdown, as the last message. Either OK message or error message.

OndroMih committed:
Runnable Embedded GlassFish improvements

* stop automatically on error when executing commands
* print Welcome message at the beginning
* print message after shutdown, as the last message. Either OK message or error message.

dmatej commented on nucleus/core/kernel/src/main/java/org/glassfish/runnablejar/UberMain.java:
```diff
@@ -213,23 +252,38 @@ private void addShutdownHook() {
             public void run() {
-                if (glassFish != null) {
-                    stopGlassFish();
+                // We restore the handlers because they are already removed from the logger in the shutdown hook
```
This is quite certainly bad idea, I will remember later why.
Btw logging should not be used for communication with the user, but we have this problem all over the sources. I already created an issue for that.

OndroMih replied:
This is not communication with the user, it’s a proper log message at the end. It can happen in Kubernetes for example, if the service is misconfigured.

what else do you suggest? System.err?

OndroMih committed:
Fix of Embedded GlassFish into panel when no listener enabled

dmatej reviewed (changes requested):
Shutdown hooks are there to do the cleanup. When they stop logging, all output properly closed, going against the stream is wrong. In extreme it can cause some damage.

If I remember well, the last message is already written by the shutdown hook, if it is not, we can fix it, but this is really a bad hack.

Handlers are also closeable (but don't implement Closeable), so they should not be used after the logging system was stopped.

I will take a look later ... embedded is bit different than Server, ie. shutdown hooks might not be executed as embedded was started inside existing JVM ... so perhaps something really has to be done, but not this way.

dmatej committed:
Code cleanup, fixed test messages

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Revided UberMain

- Better handling of System.exit and the goodbye message
- Processing errors to error codes (TODO: extend)
- enums processed completely
- Using System.console instead of STDIN for prompt
- Minimalized shutdown hook
- UberMain is now AutoCloseable, however main doesn't stop it - its purpose is
  to start it and return.
- goodbye message can be null - anything can initiate exit for any reason.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

OndroMih committed:
Merge pull request #3 from dmatej/ondromih-runnable-embedded-improvements

Revised UberMain
